platform/srf06-cc26xx/common/i2c.c

```diff
@@ -29,25 +29,43 @@
  */
 /*---------------------------------------------------------------------------*/
 /**
- * \addtogroup sensortag-cc26xx-i2c
+ * \author Andreas Urke <arurke@gmail.com>
+ *
+ * \addtogroup cc26xx-cc13xx-i2c
  * @{
  *
+ *
  * \file
- * Board-specific I2C driver for the Sensortag-CC26xx
+ * Generic I2C driver for cc26xx/13xx
  */
 /*---------------------------------------------------------------------------*/
 #include "contiki-conf.h"
 #include "ti-lib.h"
-#include "board-i2c.h"
+#include "i2c.h"
 #include "lpm.h"
 
 #include <string.h>
 #include <stdbool.h>
 /*---------------------------------------------------------------------------*/
-#define NO_INTERFACE 0xFF
+#define DEBUG 0
+#if DEBUG
+#include <stdio.h>
+#define PRINTF(...) printf(__VA_ARGS__)
+#else
+#define PRINTF(...)
+#endif
+/*---------------------------------------------------------------------------*/
+#define I2C_PIN_NOT_SET           IOID_UNUSED
+#define I2C_SLAVE_ADDRESS_NOT_SET 0x00
+#define I2C_PIN_PULL_NOT_SET      0x00
+#define I2C_DEFAULT_SPEED         I2C_SPEED_FAST
 /*---------------------------------------------------------------------------*/
-static uint8_t slave_addr = 0x00;
-static uint8_t interface = NO_INTERFACE;
+static uint8_t selected_slave_addr = I2C_SLAVE_ADDRESS_NOT_SET;
+static uint32_t selected_pin_sda = I2C_PIN_NOT_SET;
+static uint32_t selected_pin_scl = I2C_PIN_NOT_SET;
+static bool selected_speed = I2C_DEFAULT_SPEED;
+static uint32_t selected_pin_pull = I2C_PIN_PULL_NOT_SET;
```

Comment:
I wanted to be able to identify if a value was set and not (which will not be possible if using IOC_NO_IOPULL since it is a valid value). At the moment this has no implications but it seems reasonable just for the sake of consistency. If you have any arguments otherwise I am all yours :)